### PR TITLE
[GH-21565]-Add the request context and logger to all public methods in server/channels/app/audit.go

### PR DESCRIPTION
**Pull request by Kimbohlovette**
#### Summary

 In this pull request I added a request context (`rctx request.CTX`) in all public methods in `server/channels/app/audit.go`
 Passed the context to all functions calling the changed methods
 Updated the generated interfaces with `make generated` command.
 Updated all tests in the test file acordingly.

#### NOTE
In some places where modified methods were called,  there was no obvious context to pass to the methods as first argument. I created an empty request context `request.EmptyContext(a.Log())`.
Inspired by Tim Scheuemann in `server/channels/wsapi/user.go` `line:30`

Fixes https://github.com/mattermost/mattermost/issues/21565

#### Release Notes

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/system.go [hunk 1]
```diff
@@ -281,7 +281,7 @@ func getAudits(c *Context, w http.ResponseWriter, r *http.Request) {
 	audit.AddEventParameter(auditRec, "audits_per_page", c.Params.LogsPerPage)
 
 	if err := json.NewEncoder(w).Encode(audits); err != nil {
-		c.Logger.Warn("Error while writing response", mlog.Err(err))
+		c.AppContext.Logger().Warn("Error while writing response")
```
Comment (hanzei) [inline server/channels/api4/system.go:284]: While there is no different between using the two loggers, `c.Logger` is the one that is used consistently. 
```suggestion
		c.Logger.Warn("Error while writing response")
```

File: server/channels/api4/user.go [hunk 2]
```diff
@@ -2309,7 +2309,7 @@ func getUserAudits(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("audits_per_page", c.Params.LogsPerPage)
 
 	if err := json.NewEncoder(w).Encode(audits); err != nil {
-		c.Logger.Warn("Error while writing response", mlog.Err(err))
+		c.AppContext.Logger().Warn("Error while writing response")
```
Comment (hanzei) [inline server/channels/api4/user.go:2312]: Same here
```suggestion
		c.Logger.Warn("Error while writing response")
```

File: server/cmd/mattermost/commands/import.go [hunk 3]
```diff
@@ -92,10 +92,10 @@ func slackImportCmdF(command *cobra.Command, args []string) error {
 	CommandPrettyPrintln("Finished Slack Import.")
 	CommandPrettyPrintln("")
 
-	auditRec := a.MakeAuditRecord("slackImport", audit.Success)
+	auditRec := a.MakeAuditRecord(request.EmptyContext(a.Log()), "slackImport", audit.Success)
```
Comment (hanzei) [inline server/cmd/mattermost/commands/import.go:95]: Let's move the request to it's own variable at the start of the method and reuse it for all methods that need a request.

File: server/cmd/mattermost/commands/jobserver.go [hunk 4]
```diff
@@ -57,8 +58,8 @@ func jobserverCmdF(command *cobra.Command, args []string) error {
 	}
 
 	if !noJobs || !noSchedule {
-		auditRec := a.MakeAuditRecord("jobServer", audit.Success)
-		a.LogAuditRec(auditRec, nil)
+		auditRec := a.MakeAuditRecord(request.EmptyContext(a.Log()), "jobServer", audit.Success)
+		a.LogAuditRec(request.EmptyContext(a.Log()), auditRec, nil)
```
Comment (hanzei) [inline server/cmd/mattermost/commands/jobserver.go:62]: Same here

**Review by hanzei — CHANGES_REQUESTED** [review]
Nice work :+1:

**Comment by Kimbohlovette** [comment]
Okay I'll update immediately

**Comment by Kimbohlovette** [comment]
Hello @hanzei, I've applied the changes you recommended. Could you look at it?

File: server/channels/api4/system.go [hunk 5]
```diff
@@ -281,7 +281,7 @@ func getAudits(c *Context, w http.ResponseWriter, r *http.Request) {
 	audit.AddEventParameter(auditRec, "audits_per_page", c.Params.LogsPerPage)
 
 	if err := json.NewEncoder(w).Encode(audits); err != nil {
-		c.Logger.Warn("Error while writing response", mlog.Err(err))
+		c.Logger.Warn("Error while writing response")
```
Comment (hanzei) [inline server/channels/api4/system.go:284]: Whoops, that's an unrelated change

File: server/channels/api4/user.go [hunk 6]
```diff
@@ -2309,7 +2309,7 @@ func getUserAudits(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("audits_per_page", c.Params.LogsPerPage)
 
 	if err := json.NewEncoder(w).Encode(audits); err != nil {
-		c.Logger.Warn("Error while writing response", mlog.Err(err))
+		c.Logger.Warn("Error while writing response")
```
Comment (hanzei) [inline server/channels/api4/user.go:2312]: Same here

File: server/cmd/mattermost/commands/import.go [hunk 7]
```diff
@@ -57,6 +57,8 @@ func slackImportCmdF(command *cobra.Command, args []string) error {
 	}
 	defer a.Srv().Shutdown()
 
+	var rctx = request.EmptyContext(a.Log())
```
Comment (hanzei) [inline server/cmd/mattermost/commands/import.go:60]: Nit pick: It's idiomatic in go to use `:=` when possible.
```suggestion
	rctx := request.EmptyContext(a.Log())
```

File: server/cmd/mattermost/commands/import.go [hunk 8]
```diff
@@ -107,6 +109,8 @@ func bulkImportCmdF(command *cobra.Command, args []string) error {
...
```
Comment (hanzei) [inline server/cmd/mattermost/commands/import.go:112]: Same here

File: server/cmd/mattermost/commands/jobserver.go [hunk 9]
```diff
@@ -43,6 +44,8 @@ func jobserverCmdF(command *cobra.Command, args []string) error {
 
 	a.Srv().LoadLicense()
 
+	var rctx = request.EmptyContext(a.Log())
```
Comment (hanzei) [inline server/cmd/mattermost/commands/jobserver.go:47]: And here

**Review by hanzei — CHANGES_REQUESTED** [review]
We are almost there. Just a couple of comments about idiomatic go

File: server/channels/api4/system.go (hunk 5 above)
Comment (Kimbohlovette) [reply server/channels/api4/system.go:284]: Should I maintain what was there initially?

File: server/cmd/mattermost/commands/import.go (hunk 7 above)
Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/import.go:60]: Okay I get it :+1:

File: server/channels/api4/system.go (hunk 5 above)
Comment (hanzei) [reply server/channels/api4/system.go:284]: Yes, please.

Why did you remove the  error from the log message in the first place?

Comment (Kimbohlovette) [reply server/channels/api4/system.go:284]: I omitted it based on the the change that you suggested above. It did not have the `mlog.Err(err)`

Comment (hanzei) [reply server/channels/api4/system.go:284]: Oh, my bad. I didn't intent to remove it. Sorry.

Comment (Kimbohlovette) [reply server/channels/api4/system.go:284]: It's okay, I will apply those changes

**Comment by Kimbohlovette** [comment]
Hello @hanzei, I've made the necessary changes

File: server/cmd/mattermost/commands/export.go [hunk 10]
```diff
@@ -134,23 +134,22 @@ func scheduleExportCmdF(command *cobra.Command, args []string) error {
 		ctx := context.Background()
 		if timeoutSeconds > 0 {
 			var cancel context.CancelFunc
-			ctx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(timeoutSeconds))
+			_, cancel = context.WithTimeout(ctx, time.Second*time.Duration(timeoutSeconds))
 			defer cancel()
 		}
 
-		var rctx request.CTX = request.EmptyContext(a.Log())
-		rctx = rctx.WithContext(ctx)
```
Comment (hanzei) [inline server/cmd/mattermost/commands/export.go:142]: Why was `WithContext` removed?

File: server/cmd/mattermost/commands/export.go [hunk 11]
```diff
@@ -165,6 +164,8 @@ func buildExportCmdF(format string) func(command *cobra.Command, args []string)
 		}
 		defer a.Srv().Shutdown()
 
+		rctx := request.EmptyContext(a.Log())
```
Comment (hanzei) [inline server/cmd/mattermost/commands/export.go:167]: :+1:

File: server/cmd/mattermost/commands/export.go (hunk 10 above)
Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/export.go:142]: Initially the `ctx` context was declared and assigned with `context.Background()` which is just an empty context. This means that creating a context with `request.EmptyContext(a.Log())` and `request.EmptyContext(a.Log()).WithContext(ctx)` is the same thing to me

Comment (hanzei) [reply server/cmd/mattermost/commands/export.go:142]: In L137 a new `ctx` gets created using a timeout. Why is that `ctx` not longer needed?

Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/export.go:142]: Alright I saw that that is done inside the if statement. To me the new ctx which created with timeout is local to the if block.

Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/export.go:142]: Or should I bring back the line
```
rctx = rctx.WithContext(ctx)
```
as before?

File: server/cmd/mattermost/commands/export.go [hunk 12]
```diff
@@ -134,11 +136,10 @@ func scheduleExportCmdF(command *cobra.Command, args []string) error {
 		ctx := context.Background()
 		if timeoutSeconds > 0 {
 			var cancel context.CancelFunc
-			ctx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(timeoutSeconds))
+			_, cancel = context.WithTimeout(ctx, time.Second*time.Duration(timeoutSeconds))
```
Comment (hanzei) [inline server/cmd/mattermost/commands/export.go:139]: We still need the `ctx` that `context.WithTimeout`returns, right?

File: server/cmd/mattermost/commands/export.go (hunk 10 above)
Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/export.go:142]: Hello @hanzei, I've added the line of code `rctx = rctx.WithContext(ctx)`

Note that I had to change declaration at the top to `var rctx request.CTX = request.EmptyContext(a.Log())`. This is because using `rctx := request.EmptyContext(a.Log())` automatically gives `rctx` a type of `request.Context`. When I added the line 
```
rctx = rctx.WithContext(ctx)
```
I got an error because `rctx.WithContext(ctx)` returns a value of type `*request.CTX` but `rctx` given type `*request.Context`

File: server/cmd/mattermost/commands/export.go (hunk 12 above)
Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/export.go:139]: yes that is true I forgot to assign it back

Comment (Kimbohlovette) [reply server/cmd/mattermost/commands/export.go]: I've done it

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Comment by Kimbohlovette** [comment]
Hello @hanzei , @agarciamontoro  please I would like my PR to be reviewed so that I can take on another task

**Review by agarciamontoro — APPROVED** [review]
This looks great, @Kimbohlovette, thank you so much for your work!

**Comment by Kimbohlovette** [comment]
> This looks great, @Kimbohlovette, thank you so much for your work!

Thanks sir

**Comment by hanzei** [comment]
Thank you for the contribution @Kimbohlovette :+1:

**Comment by Kimbohlovette** [comment]
> Thank you for the contribution @Kimbohlovette 👍

The pleasure is mine
